src/lib/layer-manager.js

```diff
@@ -428,3 +537,33 @@ function layerName(layer) {
   }
   return !layer ? 'null layer' : 'invalid layer';
 }
+
+/**
+ * Get mouse position {x, y} relative to the container
+ */
+function getPos(event) {
+  const {srcEvent, center, rootElement} = event;
+  let x;
+  let y;
+
+  if (center) {
+    // processed pointer input, supports touch
+    x = center.x;
+    y = center.y;
+  } else if (srcEvent) {
+    // fallback
+    x = srcEvent.clientX;
+    y = srcEvent.clientY;
+  }
+  if (!Number.isFinite(x)) {
+    // srcEvent.clientX may be undefined if it's not from a mouse
+    return null;
+  }
+
+  const rect = rootElement.getBoundingClientRect();
```

Comment:
I wrote a comment here earlier today and GH seems to have swallowed it.

IIRC I suggested we use `rootElement.offsetLeft`/`Top` instead of `getBoundingClientRect()` to avoid reflows. @Pessimistress what's the use case here, was this logic added to handle the fact that touch events don't have `clientX`/`Y` or to handle situations in which the `canvas` doesn't fill the viewport?